components/match2/wikis/leagueoflegends/match_group_input_custom.lua

```diff
@@ -70,507 +47,267 @@ local CustomMatchGroupInput = {}
-			else
-				opponents[scoreIndex].placement = tonumber(opponents[scoreIndex].placement or '') or counter
-				lastPlacement = counter
-				lastScore = score or NO_SCORE
-			end
-		end
+		assert(#opponents == 2, 'Unexpected number of opponents when calculating winner')
+		return nil, tonumber(winner) or tonumber(opponents[1].score) > tonumber(opponents[2].score) and 1 or 2
 	end
-
-	return opponents, winner
 end
 
----@param tbl table[]
----@param key1 integer
----@param key2 integer
+---@param input string?
 ---@return boolean
-function CustomMatchGroupInput.placementSortFunction(tbl, key1, key2)
-	local value1 = tonumber(tbl[key1].score or NO_SCORE) or NO_SCORE
-	local value2 = tonumber(tbl[key2].score or NO_SCORE) or NO_SCORE
-	return value1 > value2
-end
-
--- Check if any opponent has a none-standard status
----@param tbl table
----@return boolean
-function CustomMatchGroupInput.placementCheckSpecialStatus(tbl)
-	return Table.any(tbl,
-		function (_, scoreinfo)
-			return scoreinfo.status ~= STATUS_SCORE and String.isNotEmpty(scoreinfo.status)
-		end
-	)
+function CustomMatchGroupInput.isNotPlayedInput(input)
+	return Table.includes(NP_INPUTS, input)
 end
 
---
--- match related functions
---
-
----@param match table
----@return table
-function MatchFunctions.getBestOf(match)
-	if tonumber(match.bestof) then
-		match.bestof = tonumber(match.bestof)
-	else
-		local mapCount = 0
-		for i = 1, MAX_NUM_GAMES do
-			if match['map'..i] then
-				mapCount = mapCount + 1
-			else
-				break
-			end
-		end
-		match.bestof = mapCount
-	end
-	return match
+---@param bestOfInput string|integer?
+---@param maps table[]
+---@return integer?
+function MatchFunctions.getBestOf(bestOfInput, maps)
+	return tonumber(bestOfInput) or #maps
 end
 
 -- Calculate the match scores based on the map results (counting map wins)
--- Only update an opponents result if it's
--- 1) Not manually added
--- 2) At least one map has a winner
----@param match table
----@return table
-function MatchFunctions.getScoreFromMapWinners(match)
-	local newScores = {}
-	local setScores = false
-
-	-- If the match has started, we want to use the automatic calculations
-	if match.dateexact then
-		if match.timestamp <= NOW then
-			setScores = true
-		end
-	end
-
-	local mapIndex = 1
-	while match['map'..mapIndex] do
-		local winner = tonumber(match['map'..mapIndex].winner)
-		if winner and winner > 0 and winner <= MAX_NUM_OPPONENTS then
-			setScores = true
-			newScores[winner] = (newScores[winner] or 0) + 1
-		end
-		mapIndex = mapIndex + 1
-	end
-
-	for index = 1, MAX_NUM_OPPONENTS do
-		if not match['opponent' .. index].score and setScores then
-			match['opponent' .. index].score = newScores[index] or 0
-		end
-	end
-
-	return match
+---@param maps {winner: integer?}[]
+---@param opponentIndex integer
+---@return integer
+function MatchFunctions.computeMatchScoreFromMapScores(maps, opponentIndex)
```

Comment:
FromMatchWinner, right?
also imo something most wikis would use, i.e. move to commons